x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStoreTests.java

```diff
@@ -1433,6 +1433,17 @@ public void testMonitoringUserRole() {
                 ),
             is(false)
         );
+
+        final String metricsPrefix = "metrics-";
+        assertNoAccessAllowed(monitoringUserRole, metricsPrefix + "system.cpu-default");
+        assertNoAccessAllowed(monitoringUserRole, metricsPrefix + "elastic_agent.filebeat-default");
+
+        assertOnlyReadAllowed(monitoringUserRole, metricsPrefix + "elasticsearch.stack_monitoring.cluster_stats-default");
+        assertOnlyReadAllowed(monitoringUserRole, metricsPrefix + "elasticsearch.ingest_pipeline-default");
+        assertOnlyReadAllowed(monitoringUserRole, metricsPrefix + "kibana.stack_monitoring.stats-default");
+        assertOnlyReadAllowed(monitoringUserRole, metricsPrefix + "logstash.stack_monitoring.node_stats-default");
+        assertOnlyReadAllowed(monitoringUserRole, metricsPrefix + "beats.stack_monitoring.stats-default");
```

Comment:
All the tests include something matching the `.*`, but to prevent it from being changed to `.+` in the future, can you add an assert for `metricsPrefix + "elasticsearch"` (or some other equivalent example)?